state/aws/dynamodb/dynamodb.go

```diff
@@ -393,6 +393,61 @@ func (d *StateStore) parseTTL(req *state.SetRequest) (*int64, error) {
 	return nil, nil
 }
 
+// Multi performs a transactional operation. succeeds only if all operations succeed, and fails if one or more operations fail.
+func (d *StateStore) Multi(ctx context.Context, request *state.TransactionalStateRequest) error {
+	twinput := &dynamodb.TransactWriteItemsInput{
+		TransactItems: []*dynamodb.TransactWriteItem{},
+	}
+
+	// Dedup ops where the last operation with a matching Key takes precedence
+	txs := map[string]int{}
+	for i, o := range request.Operations {
+		txs[o.GetKey()] = i
```

Comment:
Just trying to think if it is safe to do dedup of ops here and if yes, then why we have been not doing similar in other state stores as well.

Reply:
@DeepanshuA 

Good question!
I added this logic due to a DynamoDB error I was running into while with the conformance tests. Here is the error 
```
ValidationException: Transaction request cannot include multiple operations on one item
```
The error was coming due to duplicate `keys` within a set of operations.
As part of my investigations, I had conversations with folks that know DynamoDB well and came to the conclusion that adding this here would both avoid getting the error and should not cause any bad side effects.

If this turns out to be a general practice, maybe a similar logic can be added at a higher level in the State Multi Transaction execution.

Reply:
Ok if this is needed for a DynamoDB-specific behavior, please add a comment explaining that this component is the only one behaving that way, and why this is necessary. In the docs, we should mention this behavior too.

It's good our conf tests include this behavior so we could validate it. I find reassurance in the fact that we don't believe many users actually rely on Multi to perform multiple operations on the same key.

Reply:
Yup, makes sense then. Agree with @ItalyPaleAle for comment and docs.

Reply:
Absolutely! I was planning to update the AWS DynamoDB Component docs indicating this behavior.